test/functional/services/flyout.ts

```diff
@@ -31,20 +31,17 @@ export class FlyoutService extends FtrService {
   }
 
   public async ensureAllClosed(): Promise<void> {
-    const flyoutElements = await this.find.allByCssSelector('.euiFlyout');
-
-    if (!flyoutElements.length) {
-      return;
-    }
-
-    for (let i = 0; i < flyoutElements.length; i++) {
-      const closeBtn = await flyoutElements[i].findByCssSelector('[aria-label*="Close"]');
-      await closeBtn.click();
-    }
-
-    await this.retry.waitFor(
-      'all flyouts to be closed',
-      async () => (await this.find.allByCssSelector('.euiFlyout')).length === 0
-    );
```

Comment:
This logic misses the case when some flyout was not successfully closed and will stuck until `retry.waitFor` fails with timeout. I re-wrote the function with the following assumption that flayouts may appear asynchronously
- wait for up to 2500 ms and close all found ones
- wait for up to 1000 ms if there still any old/new flayouts displayed
- retry from start if still any displayed